backend/api/Services/ActivityService.cs

```diff
@@ -0,0 +1,122 @@
+
+            return activityInstance;
+        }
+
+        public IList<PimsActivityInstance> GetAllByResearchFileId(long researchFileId)
+        {
+            _logger.LogInformation("Getting activities by research id {researchFileId}", researchFileId);
+            this.User.ThrowIfNotAuthorized(Permissions.NoteView);
+
+            // _user.ThrowIfNotAuthorized(Permissions.ResearchFileView);
+            var activityInstances = _activityRepository.GetAllByResearchFileId(researchFileId);
+
+            return activityInstances;
+        }
+
+        public IList<PimsActivityTemplate> GetAllActivityTemplates()
+        {
+            _logger.LogInformation("Getting activity templates");
+            this.User.ThrowIfNotAuthorized(Permissions.NoteView);
+
+            // _user.ThrowIfNotAuthorized(Permissions.ResearchFileView);
+            var activtyTemplates = _activityTemplateRepository.GetAllActivityTemplates();
+            return activtyTemplates;
+        }
+
+        public PimsActivityInstance Add(PimsActivityInstance instance)
+        {
+            _logger.LogInformation("Adding activity instance ...");
+            this.User.ThrowIfNotAuthorized(Permissions.NoteAdd);
+
+            // _user.ThrowIfNotAuthorized(Permissions.ResearchFileAdd);
+            var newActivityInstance = _activityRepository.Add(instance);
+            _activityRepository.CommitTransaction();
+            return _activityRepository.GetById(newActivityInstance.ActivityInstanceId);
+        }
+
+        public PimsActivityInstance Update(PimsActivityInstance model)
+        {
+            _logger.LogInformation("Updating activity instance ...", model);
+            this.User.ThrowIfNotAuthorized(Permissions.NoteEdit);
+            ValidateVersion(model.ActivityInstanceId, model.ConcurrencyControlNumber);
+
+            // _user.ThrowIfNotAuthorized(Permissions.ResearchFileAdd);
+            var newActivityInstance = _activityRepository.Update(model);
+            _activityRepository.CommitTransaction();
+            return _activityRepository.GetById(newActivityInstance.ActivityInstanceId);
+        }
+
+        public void Delete(long activityId)
+        {
+            _logger.LogInformation("Deleting activity instance ...", activityId);
+            this.User.ThrowIfNotAuthorized(Permissions.NoteDelete);
+
+            // _user.ThrowIfNotAuthorized(Permissions.ResearchFileAdd);
+            var activityDocuments = _documentService.GetActivityDocuments(activityId);
+            foreach (PimsActivityInstanceDocument activityInstanceDocument in activityDocuments)
+            {
+                _documentService.DeleteActivityDocumentAsync(activityInstanceDocument);
+            }
+
+            var notes = _noteService.GetNotes(Constants.NoteType.Activity, activityId);
+            foreach (PimsNote note in notes)
+            {
+                _noteService.DeleteNote(Constants.NoteType.Activity, note.Id);
+            }
```

Comment:
Just to confirm. This is to delete the notes for the given activity right?

Reply:
yes, when an activity is deleted all corresponding notes should be deleted.